src/include/cart/api.h

```diff
@@ -706,227 +705,200 @@ crt_ep_abort(crt_endpoint_t *ep);
+				if (proc_op == CRT_PROC_DECODE)                                    \
+					ptr->CRT_GEN_GET_NAME(seq).ca_count = i;                   \
+				goto CRT_GEN_FAIL_LABEL(seq);                                      \
+			}                                                                          \
+		}                                                                                  \
+		if (proc_op == CRT_PROC_FREE)                                                      \
+			D_FREE(*e_ptrp);                                                           \
 	} while (0);
 
-#define CRT_GEN_FAIL_ARRAY(ptr, seq)					\
-	CRT_GEN_FAIL_LABEL(seq):					\
-	if (proc_op == CRT_PROC_DECODE) {				\
-		CRT_GEN_GET_TYPE(seq) **e_ptrp = &ptr->CRT_GEN_GET_NAME(seq).ca_arrays;\
-		CRT_GEN_GET_TYPE(seq) *e_ptr = ptr->CRT_GEN_GET_NAME(seq).ca_arrays; \
-		uint64_t count = ptr->CRT_GEN_GET_NAME(seq).ca_count;	\
-		uint64_t i;						\
-		/* process the elements of array */			\
-		for (i = 0; i < count && e_ptr != NULL; i++)		\
-			(void)CRT_GEN_GET_FUNC(seq)(proc, CRT_PROC_FREE,\
-						    &e_ptr[i]);		\
-		D_FREE(*e_ptrp);					\
-		ptr->CRT_GEN_GET_NAME(seq).ca_count = 0;		\
+#define CRT_GEN_FAIL_ARRAY(ptr, seq)                                                               \
+	CRT_GEN_FAIL_LABEL(seq) : if (proc_op == CRT_PROC_DECODE)                                  \
+	{                                                                                          \
+		CRT_GEN_GET_TYPE(seq) **e_ptrp = &ptr->CRT_GEN_GET_NAME(seq).ca_arrays;            \
+		CRT_GEN_GET_TYPE(seq) *e_ptr   = ptr->CRT_GEN_GET_NAME(seq).ca_arrays;             \
+		uint64_t count                 = ptr->CRT_GEN_GET_NAME(seq).ca_count;              \
+		uint64_t i;                                                                        \
+		/* process the elements of array */                                                \
+		for (i = 0; i < count && e_ptr != NULL; i++)                                       \
+			(void)CRT_GEN_GET_FUNC(seq)(proc, CRT_PROC_FREE, &e_ptr[i]);               \
+		D_FREE(*e_ptrp);                                                                   \
+		ptr->CRT_GEN_GET_NAME(seq).ca_count = 0;                                           \
 	}
 
-#define CRT_GEN_PROC_RAW(ptr, seq)					\
-	rc = crt_proc_memcpy(proc, proc_op, &ptr->CRT_GEN_GET_NAME(seq),\
-			     sizeof(CRT_GEN_GET_TYPE(seq)));		\
-	if (unlikely(rc))						\
+#define CRT_GEN_PROC_RAW(ptr, seq)                                                                 \
+	rc = crt_proc_memcpy(proc, proc_op, &ptr->CRT_GEN_GET_NAME(seq),                           \
+			     sizeof(CRT_GEN_GET_TYPE(seq)));                                       \
+	if (unlikely(rc))                                                                          \
 		goto CRT_GEN_FAIL_LABEL(seq);
 
-#define CRT_GEN_FAIL_RAW(ptr, seq)					\
-	CRT_GEN_FAIL_LABEL(seq):;
+#define CRT_GEN_FAIL_RAW(ptr, seq) CRT_GEN_FAIL_LABEL(seq) :;
```

Comment:
(style)  spaces required around that ':' (ctx:WxO)
(style)  Macros with multiple statements should be enclosed in a do - while loop
(style)  macros should not use a trailing semicolon